libr/core/cmd_anal.c

```diff
@@ -3823,16 +3823,29 @@ R_API int r_core_esil_step(RCore *core, ut64 until_addr, const char *until_expr,
 	RAnalEsil *esil = core->anal->esil;
 	const char *name = r_reg_get_name (core->anal->reg, R_REG_NAME_PC);
 	bool breakoninvalid = r_config_get_i (core->config, "esil.breakoninvalid");
+	int esiltimeout = r_config_get_i (core->config, "esil.timeout");
+	ut64 startTime;
 	if (!esil) {
 		initializeEsil (core);
 	}
+	if (esiltimeout) {
+		startTime = r_sys_now();
+	}
 	ut64 addr = r_reg_getv (core->anal->reg, name);
 	r_cons_break_push (NULL, NULL);
 repeat:
 	if (r_cons_is_breaked ()) {
 		eprintf ("[+] ESIL emulation interrupted at 0x%08" PFMT64x "\n", addr);
 		return_tail (0);
 	}
+	//Break if we have exceeded esil.timeout
+	if (esiltimeout) {
+		ut64 elapsedTime = r_sys_now () - startTime;
+		if (elapsedTime >= (esiltimeout * CLOCKS_PER_SEC)) {
```

Comment:
dont use this CLOCKS_PER_SEC constant. in fact r_sys_now() do a <<= 20 for the seconds, so you can shift that value back to drop the lower timing info. We probably want to have a macro or a function for this instead of depending on the posix stuff which is unrelated here because r_sys_now() uses gettimeofday()

Reply:
Fixed the use of the `CLOCKS_PER_SEC` const, now just shifting back. Didn't spot that, thanks.